Typescript 4.5.4 does not like sequelize-ESM imports in v6.12

My understanding of https://github.com/sequelize/sequelize/pull/13689, included in v6.12 is, that sequelize is not any more exported just in CommonJS, but can be used also as ESM-imports.  Thus, the code in v6.11
```JavaScript
import sequelize_ from 'sequelize'
const {DataTypes, Model, Op} = sequelize_ //this is CommonJS hack
```
can be changed in 6.12 to `import {DataTypes, Model, Op} from 'sequelize'`.

Typescript 4.5.4 fails on the above code with
```
$ node_modules/.bin/tsc
node_modules/sequelize/lib/operators.ts:524:1 - error TS1203: Export assignment cannot be used when targeting ECMAScript modules. Consider using 'export default' or another module format instead.

524 export = Op;
    ~~~~~~~~~~~~

Found 1 error.
```
My understanding is, that Typescript is only supposed to verify the types in sequelize by parsing the code in the `node_modules/sequelize/types` directory, and not loose time to verify each file in `node_modules/sequelize/lib/`.  But `types/index.d.ts` contains `import * as Op from "../lib/operators";` so my assumption is not correct.  Likewise for `node_modules/sequelize/types/lib/model.d.ts:import * as Op from '../../lib/operators';`


## Bug Report Checklist

### How does this problem relate to dialects?

- [x] I think this problem happens regardless of the dialect.

### Would you be willing to resolve this issue by submitting a Pull Request?
- [x] No, I don't have the time and I wouldn't even know how to start.

--- a/lib/operators.ts
+++ b/lib/operators.ts
@@ -521,4 +521,9 @@ const Op: OpTypes = {
   match: Symbol.for('match')
 } as OpTypes;
 
-export = Op;
+export default Op;
+
+// https://github.com/sequelize/sequelize/issues/13791
+// remove me in v7: kept for backward compatibility as `export default Op` is
+// transpiled to `module.exports.default` instead of `module.exports`
+module.exports = Op;

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,6 +1,6 @@
 import DataTypes = require("./lib/data-types");
 import Deferrable = require("./lib/deferrable");
-import Op = require("../lib/operators");
+import Op from "../lib/operators";
 import QueryTypes = require("./lib/query-types");
 import TableHints = require("./lib/table-hints");
 import IndexHints = require("./lib/index-hints");

--- a/types/lib/model.d.ts
+++ b/types/lib/model.d.ts
@@ -9,7 +9,7 @@ import { Sequelize, SyncOptions } from './sequelize';
 import { LOCK, Transaction } from './transaction';
 import { Col, Fn, Literal, Where } from './utils';
 import { SetRequired } from '../type-helpers/set-required'
-import Op = require('../../lib/operators');
+import Op from '../../lib/operators';
 
 export interface Logging {
   /**

